src/Features/Core/Portable/PullMemberUp/QuickAction/ClassPullerWithQuickAction.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.  
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+
+namespace Microsoft.CodeAnalysis.PullMemberUp.QuickAction
+{
+    internal class ClassPullerWithQuickAction : AbstractMemberPullerWithQuickAction
+    {
+        protected override bool IsDeclarationAlreadyInTarget(INamedTypeSymbol targetSymbol, ISymbol userSelectedNodeSymbol)
+        {
+            if (userSelectedNodeSymbol is IFieldSymbol fieldSymbol)
+            {
+                // If a member whose name is same as the selected one, then don't provide the refactoring option
+                return targetSymbol.GetMembers(fieldSymbol.Name).Any(member => member.Kind == SymbolKind.Field);
```

Comment:
it would be worthwhile ot explain why.

Reply:
code does'nt match comment.  you say "If a member whose name is same as the selected one".  but you're only checking fields due to the `==` check.  Either make code match comment, or comment amtch code.